bindings/py/tests/algorithms/SDR_test.py

```diff
@@ -0,0 +1,242 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2019, David McDougall
+# The following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""Unit tests for SDR."""
+
+import pickle
+import numpy as np
+import unittest
+
+from nupic.bindings.algorithms import SDR
+
+class SdrTest(unittest.TestCase):
+    def testConstructor(self):
+        A = SDR((103,))
+        B = SDR((100, 100, 1))
+        assert(tuple(B.dimensions) == (100, 100, 1))
+        C = SDR((2, 4, 5, 1,1,1,1, 3))
```

Comment:
this is a bit confusing, what is it, initialization by given set of active bits? Add a doc please

Reply:
Ok, In the next commit I will change this to:
```
        # Test crazy dimensions, also test keyword arguments.
        C = SDR(dimensions = (2, 4, 5, 1,1,1,1, 3) )
```

Reply:
so it's an 8-dim SDR?